[functorch] Fix dangling impls

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #85299
* #85152
* #85151

Our dangling impls were:
- positive_ (the in-place op just never existed)
- unique (something happened to this op, maybe it was renamed)

Test Plan:
- `import functorch; torch._C._dispatch_find_dangling_impls`
- It's difficult to write a test for this because the number of dangling
impls depends on if `test_dispatch` has been run already or not
(test_dispatch adds a dangling impl)
- Can't remove the torchdynamo skip for this yet either

zou3519 committed:
[functorch] Fix dangling impls

Our dangling impls were:
- positive_ (the in-place op just never existed)
- unique (something happened to this op, maybe it was renamed)

Test Plan:
- `import functorch; torch._C._dispatch_find_dangling_impls`
- It's difficult to write a test for this because the number of dangling
impls depends on if `test_dispatch` has been run already or not
(test_dispatch adds a dangling impl)
- Can't remove the torchdynamo skip for this yet either

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] Fix dangling impls"


Our dangling impls were:
- positive_ (the in-place op just never existed)
- unique (something happened to this op, maybe it was renamed)

Test Plan:
- `import functorch; torch._C._dispatch_find_dangling_impls`
- It's difficult to write a test for this because the number of dangling
impls depends on if `test_dispatch` has been run already or not
(test_dispatch adds a dangling impl)
- Can't remove the torchdynamo skip for this yet either

[ghstack-poisoned]

samdow commented on test/test_dispatch.py:
```diff
@@ -759,6 +759,10 @@ def test_overwrite_math(self):
+    # function but not a def() for it. This is usually a bug, e.g. someone
+    # misspelled an operator name, or someone registered an impl for an op that
+    # no longer exists
     @skipIfTorchDynamo("Installing functorch reveals a dangling impl - aten::postive_")
```
Do we want to get rid of this skip or is this not enough to fix it?

zou3519 replied:
I'm confused by all of the dynamo pinning mechanisms so my plan was to merge this PR, wait two days, and then remove the skips. This PR should be enough to fix the skip

zou3519 committed:
Update on "[functorch] Fix dangling impls"


Our dangling impls were:
- positive_ (the in-place op just never existed)
- unique (something happened to this op, maybe it was renamed)

Test Plan:
- `import functorch; torch._C._dispatch_find_dangling_impls`
- It's difficult to write a test for this because the number of dangling
impls depends on if `test_dispatch` has been run already or not
(test_dispatch adds a dangling impl)
- Can't remove the torchdynamo skip for this yet either

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -g
